### PR TITLE
Proto file changes for identity and metaservice and volume interfaces

### PR DESCRIPTION
**Pull request by sushanthakumar**
**What type of PR is this?**
 /kind new feature

**What this PR does / why we need it**:
This PR adds the proto file and respective code related to identity and metatada provider interfaces

**Which issue(s) this PR fixes**:
Fixes: #24

**Test Report Added?**:
 /kind NOT-TESTED
Will be tested through metaservice and nfs provider modules

**Test Report**:

**Special notes for your reviewer**:
Proto file for reference : 
[providerservice.proto.txt](https://github.com/soda-cdm/kahu/files/8886325/providerservice.proto.txt)

### CODE REVIEW

**Comment by sushanthakumar** [comment]
Hi @AmitRoushan @PravinRanjan10 @joseph-v @doubletao318 pls help to review this PR

**Review by joseph-v — APPROVED** [review]
LGTM

**Comment by AmitRoushan** [comment]
IMO, The PR also needs recepie for interface file geneation

**Review by AmitRoushan — CHANGES_REQUESTED** [review]
IMO, The PR also needs recepie for interface file geneation

**Comment by sushanthakumar** [comment]
> IMO, The PR also needs recepie for interface file geneation

Updated

**Comment by sushanthakumar** [comment]
@AmitRoushan @PravinRanjan10 @joseph-v , updated review comments, pls check and approve

**Review by PravinRanjan10 — APPROVED** [review]
LGTM

**Review by joseph-v — APPROVED** [review]
LGTM

File: providers/proto/providerservice.proto [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+  map <string, string> manifest = 3;
+}
+
+message GetProviderCapabilitiesRequest{}
+
+message GetProviderCapabilitiesResponse{
+  repeated ProviderCapability capabilities = 1;
+}
+
+message ProviderCapability {
```
Comment (AmitRoushan) [inline providers/proto/providerservice.proto:36]: Add descriptions for attributes

File: providers/Makefile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+$(PROVIDER_SERVICE_INTERFACE): $(PROVIDER_SERVICE_PROTO)
+	($(PROTOC) $(INCLUDE) --go_out=$(GO_OUT) --plugin=protoc-gen-go=$(PROTOC_GEN_GO) $(PROVIDER_SERVICE_PROTO))
+
+$(PROVIDER_SERVICE_GO): $(PROVIDER_SERVICE_INTERFACE)
+
+build: $(PROVIDER_SERVICE_GO)
+
+clean:
+	rm $(PROVIDER_SERVICE_INTERFACE)
+
```
Comment (AmitRoushan) [inline providers/Makefile:40]: Please link the Makefile with Main Makefile

**Review by AmitRoushan — CHANGES_REQUESTED** [review]

File: providers/Makefile (hunk 2 above)
Comment (sushanthakumar) [reply providers/Makefile:42]: Updated

File: providers/proto/providerservice.proto (hunk 1 above)
Comment (sushanthakumar) [reply providers/proto/providerservice.proto:46]: Updated

**Review by AmitRoushan — APPROVED** [review]
LGTM

**Review by PravinRanjan10 — APPROVED** [review]
LGTM

**Pull request merged**